cmd/api-mock/main.go

```diff
@@ -30,9 +36,55 @@ func Health(w http.ResponseWriter, r *http.Request) {
 	w.WriteHeader(http.StatusOK)
 }
 
+type ClustersEndpoint struct {
+	data    map[ClustersKey]string
+	dataMux *sync.RWMutex
```

Comment:
```suggestion
	dataMux sync.RWMutex
```
When declared as a pointer, a new mutex needs to be allocated in `NewClustersEndpoint`.
Declaring it without indirection does not require this extra step:
> The zero value for a RWMutex is an unlocked mutex.
https://pkg.go.dev/sync#RWMutex

In general `Mutex` objects are not indirected by pointer, because you never pass them around on their own (or deadlocks and overall hilarity is ensured).
A mutex is usually passed as part of a struct next to the memory it is guarding.

So you would pass around a pointer to an instance of `ClustersEndpoint`. Passing around the value will copy the whole container which will lead to very strange behavior when the struct contains non-pointers.